src/kOS/Module/kOSProcessor.cs

```diff
@@ -818,19 +871,26 @@ private void ProcessorModeChanged()
         {
             switch (ProcessorMode)
             {
-            case ProcessorModes.READY:
-                shared.VolumeMgr.SwitchTo(SafeHouse.Config.StartOnArchive ? (Volume)Archive : HardDisk);
-                if (shared.Cpu != null) shared.Cpu.Boot();
-                if (shared.Interpreter != null) shared.Interpreter.SetInputLock(false);
-                if (shared.Window != null) shared.Window.IsPowered = true;
-                break;
-
-            case ProcessorModes.OFF:
-            case ProcessorModes.STARVED:
-                if (shared.Interpreter != null) shared.Interpreter.SetInputLock(true);
-                if (shared.Window != null) shared.Window.IsPowered = false;
-                if (shared.BindingMgr != null) shared.BindingMgr.UnBindAll();
-                break;
+                case ProcessorModes.READY:
+                    if (SafeHouse.Config.StartOnArchive)
+                    {
+                        shared.VolumeMgr.SwitchTo(Archive);
+                    }
+                    else
+                    {
+                        shared.VolumeMgr.SwitchTo(HardDisk);
+                    }
```

Comment:
Okay that sounded a bit snarky.  Sorry about that.  I just wanted to verify it was your intention to not actually change the algorithm meaningfully, because this looks like it's the same exact thing just expressed differently.